narwhal/primary/src/certificate_fetcher.rs

```diff
@@ -261,38 +262,33 @@ impl CertificateFetcher {
             self.targets.values().max().unwrap_or(&0),
             gc_round
         );
-        self.fetch_certificates_task.push(
-            spawn_monitored_task!(async move {
-                let _scope = monitored_scope("CertificatesFetching");
-                state
-                    .metrics
-                    .certificate_fetcher_inflight_fetch
-                    .with_label_values(&[&committee.epoch.to_string()])
-                    .inc();
+        self.fetch_certificates_task.spawn(async move {
```

Comment:
Can we keep the `spawn_monitored_task!()`?

Reply:
If that's important I guess we could spawn a second task inside , but then we eventually spawn 2 tasks when one would suffice , but probably not a big deal

Reply:
Ok, I was using it to check when fetching is active but I can probably rely on `monitored_scope()` instead.

Reply:
Didn't know about `monitored_scope` - that's for sharing the info ! If that would do the trick then let's use the `monitored_scope` info

Reply:
There's also monitored_future that doesn't involve spawning, it just creates the future IIRC

Reply:
Ah very good point @bmwill , thanks, totally forgot about the `monitored_future`